elastic/shared/runners/remote_cluster.py

```diff
@@ -1,143 +1,183 @@
-    * `remote-cluster`: (mandatory) the name of the remote cluster specified in `target-hosts` parameters
+    Runner that configures all other clusters specified in `target-hosts` to follow specified indices from the 'source-cluster'
+    * `source-cluster`: (mandatory) the name of the cluster specified in `target-hosts` parameters from which to follow indices.
     * `index`: (mandatory) the pattern of indices to be replicated
     * `request-timeout`: (optional) the timeout of this runner. Defaults to 7200
-    * `remote-connection-name`: (optional) the name of the remote connection;
-    *                           defaults to the name of the remote cluster
     """
 
     multi_cluster = True
 
     def __init__(self):
         super().__init__()
 
-    async def __call__(self, multi_es, params):
-        end_request_timeout = time.process_time() + params.get("request-timeout", 7200)
-        required_licenses = ["trial", "platinum", "enterprise"]
-
-        def request_timeout():
-            return math.ceil(end_request_timeout - time.process_time())
-
-        remote_es = multi_es[params["remote-cluster"]]
-        local_es = multi_es[params["local-cluster"]]
+    def request_timeout(self, timeout):
+        end_request_timeout = time.process_time() + timeout
+        return math.ceil(end_request_timeout - time.process_time())
 
-        if "remote-connection-name" in params:
-            remote_cluster = params["remote-connection-name"]
-        else:
-            remote_info_resp = await remote_es.info()
-            remote_cluster = remote_info_resp["cluster_name"]
+    async def _follow_indices(self, params, source_cluster_name, source_cluster_client, following_cluster_name, following_cluster_client):
+        request_timeout = self.request_timeout(params.get("request-timeout", 7200))
+        required_licenses = ["trial", "platinum", "enterprise"]
 
-        remote_license = await remote_es.license.get()
-        local_license = await local_es.license.get()
-        local_license_type = local_license.get("license", {}).get("type")
-        remote_license_type = remote_license.get("license", {}).get("type")
+        following_license = await following_cluster_client.license.get()
+        source_license = await source_cluster_client.license.get()
+        source_license_type = source_license.get("license", {}).get("type")
+        following_license_type = following_license.get("license", {}).get("type")
 
-        if local_license_type not in required_licenses or remote_license_type not in required_licenses:
+        if source_license_type not in required_licenses or following_license_type not in required_licenses:
             raise BaseException(
-                f"Cannot use license type(s) [{local_license_type}, {remote_license_type}] "
+                f"Cannot use license type(s) [{source_license_type}, {following_license_type}] "
```

Comment:
In actuality here, one or the other could be permitted, so I think this needs to be split out. It would also be helpful to spit out the cluster name, even though I don't think it's likely we'll use this track in a scenario with a mixture of follower cluster licenses

Reply:
👍 - addressed in https://github.com/elastic/rally-tracks/pull/333/commits/d30cf62b32be11beb4b1dc3d9d0de16062b6c83e